--- grub.cfg.orig       2016-06-11 08:19:49.621767413 -0400
+++ grub.cfg    2016-06-11 08:20:29.945765450 -0400
@@ -2,19 +2,19 @@ set timeout=30

 menuentry 'Android-x86 VER Live' --class android-x86 {
        search --file --no-floppy --set=root /system.sfs
-       linux /kernel CMDLINE quiet DATA=
+       linux /kernel CMDLINE quiet acpi_sleep=s3_bios,s3_mode DATA=
        initrd /initrd.img
 }

 menuentry 'Android-x86 VER DEBUG mode' --class android-x86 {
        search --file --no-floppy --set=root /system.sfs
-       linux /kernel CMDLINE DATA= DEBUG=2
+       linux /kernel CMDLINE DATA= acpi_sleep=s3_bios,s3_mode DEBUG=2
        initrd /initrd.img
 }

 menuentry 'Android-x86 VER Installation' --class android-x86 {
        search --file --no-floppy --set=root /system.sfs
-       linux /kernel CMDLINE DEBUG= INSTALL=1
+       linux /kernel CMDLINE DEBUG= acpi_sleep=s3_bios,s3_mode INSTALL=1
        initrd /initrd.img
 }
